srcpkgs/efibootmgr/files/kernel.d/efibootmgr.post-install

```diff
@@ -28,18 +28,25 @@ major_version=$(echo $PKGNAME | cut -c 6-)
 # look for previous entry for this major kernel version
 existing_entry=$(efibootmgr | grep "Void Linux with kernel ${major_version}")
 
-# get the boot order
-# this is required because when in the next step the existing entry is removed,
-# it is also removed from the order so it needs to be restored later
-bootorder=$(efibootmgr |grep "BootOrder: " |cut -c 12-)
-
 # if existing, remove it
 if [ "$existing_entry" != "" ]; then
-  /etc/kernel.d/post-remove/50-efibootmgr $PKGNAME
+	/etc/kernel.d/post-remove/50-efibootmgr $PKGNAME
 fi
 
-# create the new entry
-efibootmgr -qc $args -L "Void Linux with kernel ${major_version}" -l /vmlinuz-${VERSION} -u "${OPTIONS}"
-
-# restore the boot order
-efibootmgr -qo $bootorder
+# if using strict efi suffix
+if [ "x${USE_STRICT_EFI_SUFFIX}" = x1 ]; then
+	# rename vmlinuz
+	if [ -f /boot/vmlinuz-${VERSION} ]; then
+		mv -f /boot/vmlinuz-${VERSION} /boot/vmlinuz-${VERSION}.efi
```

Comment:
I'm a bit wary of moving the kernel around. What about copying the kernel and the initrd to the actual ESP partition. Note that grub expects the ESP to be mounted in /boot/EFI, but efibootmgr expects it to be mounted in /boot, hence they cannot easily coexist.

Copying the kernel to the ESP partition (with the .efi suffix) would solve both problems.

Reply:
+1 for copying the kernel rather than renaming.

Reply:
I'm on it.

Reply:
>Note that grub expects the ESP to be mounted in /boot/EFI, but efibootmgr expects it >to be mounted in /boot, hence they cannot easily coexist.
Where do you know this from? I've not encountered any such problems and the default placement of the non .efi kernel is still on `/boot`. 
>Copying the kernel to the ESP partition (with the .efi suffix) would solve both problems.
Could you explain it in more details?

Reply:
I correct myself: grub _can_ be installed with ESP on `/boot`.

However the installation guide recommends/assumes the ESP is on `/boot/efi` (see https://docs.voidlinux.org/installation/guides/chroot.html#create-a-new-root-and-mount-filesystems and https://docs.voidlinux.org/installation/guides/chroot.html#installing-grub)

When I wanted to switch to efibootmgr from grub, I had to move the contents of the `/boot` directory into the ESP, and then mount the ESP on `/boot`. Having the ESP on `/boot` is a requirement (in conflict with the above recommendation).

It would be nicer if efibootmgr can work with the ESP on the reocmmended location `/boot/efi`, even better if this can be configured or detected.

BTW, in case it's useful here's a silly way of figuring out where's the efi partition:
```
mount -t vfat | sed -nE 's|^([^ ]*) on (/boot(/[^ ]*)?) type vfat .*$|\1 \2|p'
```

I suppose a sensible location for the kernel+initrd might be `\EFI\void\vmlinuz-$VERSION.efi` and `\EFI\void\initramfs-$VERSION.img` on the ESP, would that work for you?

Reply:
This is getting really complicated, I don't know much about how efibootmgr works, or where the efi files should be located. And I really don't want this pr to fix all the issues that efibootmgr currently has. I created this pr because I had an issue with my laptop, and this pr fixes that issue, and only that issue.